syntax = "proto3";

option go_package = "github.com/ujinjinjin/user_service";

package user_service;

service UserService {
    rpc CreateUser(CreateUserRequest) returns (CreateUserReply);
}

// User info
message CreateUserRequest {
    // First name
    string first_name = 1;

    // Last name
    string last_name = 2;

    // Middle name
    string middle_name = 3;

    // Email address
    string email = 4;
}

message CreateUserReply {
    // User ID
    int32 id = 1;
}